### PR TITLE
clean up workflow and optional bandpass

### PR DESCRIPTION
**Pull request by damaggu**

### CODE REVIEW

File: samri/pipelines/preprocess.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from os import path, listdir, getcwd, remove
-from samri.pipelines.extra_functions import get_data_selection, get_bids_scan, write_bids_metadata_file, write_bids_events_file, force_dummy_scans, BIDS_METADATA_EXTRACTION_DICTS
+from samri.pipelines.extra_functions import get_data_selection, get_bids_scan, write_bids_metadata_file,write_bids_events_file ,write_events_file, force_dummy_scans, BIDS_METADATA_EXTRACTION_DICTS
```
Comment (TheChymera) [inline samri/pipelines/preprocess.py]: `write_events_file` is deprecated, we now use `write_bids_events_file`, and the former function has been removed from the package. If this import worked on your machine, you should sync your SAMRI again.

File: samri/pipelines/preprocess.py [hunk 2]
```diff
@@ -149,22 +153,17 @@ def bruker(bids_base, template,
...
-	#infosource.iterables = [('subject_session', subjects_sessions)]
-
 	get_f_scan = pe.Node(name='get_f_scan', interface=util.Function(function=get_bids_scan,input_names=inspect.getargspec(get_bids_scan)[0], output_names=['scan_path','scan_type','task', 'nii_path', 'nii_name', 'file_name', 'events_name', 'subject_session']))
 	get_f_scan.inputs.ignore_exception = True
 	get_f_scan.inputs.data_selection = data_selection
 	get_f_scan.inputs.bids_base = bids_base
 	get_f_scan.iterables = ("ind_type", func_ind)
 
-	dummy_scans = pe.Node(name='dummy_scans', interface=util.Function(function=force_dummy_scans,input_names=inspect.getargspec(force_dummy_scans)[0], output_names=['out_file','deleted_scans']))
+	dummy_scans = pe.Node(name='dummy_scans', interface=util.Function(function=force_dummy_scans,input_names=inspect.getargspec(force_dummy_scans)[0], output_names=['out_file']))
```
Comment (TheChymera) [inline samri/pipelines/preprocess.py]: the `force_dummy_scans` function now returns 2 values. if you instead use a length-1 list of `output_names` you will end up passing a 2-tuple to the next node.

File: samri/pipelines/preprocess.py [hunk 3]
```diff
@@ -174,20 +173,11 @@ def bruker(bids_base, template,
...
-	bids_stim_filename = pe.Node(name='bids_stim_filename', interface=util.Function(function=bids_naming,input_names=inspect.getargspec(bids_naming)[0], output_names=['filename']))
-	bids_stim_filename.inputs.suffix = "events"
-	bids_stim_filename.inputs.extension = ".tsv"
-	bids_stim_filename.inputs.metadata = data_selection
-
-	events_file = pe.Node(name='events_file', interface=util.Function(function=write_bids_events_file,input_names=inspect.getargspec(write_bids_events_file)[0], output_names=['out_file']))
-	#events_file.inputs.very_nasty_bruker_delay_hack = very_nasty_bruker_delay_hack
-	#if not (strict or verbose):
-	#	events_file.inputs.ignore_exception = True
+	events_file = pe.Node(name='events_file', interface=util.Function(function=write_bids_events_file,input_names=inspect.getargspec(write_events_file)[0], output_names=['out_file']))
```
Comment (TheChymera) [inline samri/pipelines/preprocess.py]: the function and the input names should be surced from the same function. Also, you are using the deprecated `write_events_file` function. Please apply your changes on top of the newest commit: 27afb472b5112910c2f70ad05f85818accba4dcf

File: samri/pipelines/preprocess.py [hunk 4]
```diff
@@ -174,20 +173,11 @@ def bruker(bids_base, template,
 	else:
 		bandpass.inputs.lowpass_sigma = tr
 
-	#bids_filename = pe.Node(name='bids_filename', interface=util.Function(function=sss_filename,input_names=inspect.getargspec(sss_filename)[0], output_names=['filename']))
-	bids_filename = pe.Node(name='bids_filename', interface=util.Function(function=bids_naming,input_names=inspect.getargspec(bids_naming)[0], output_names=['filename']))
-	bids_filename.inputs.metadata = data_selection
-
-	#bids_stim_filename = pe.Node(name='bids_stim_filename', interface=util.Function(function=sss_filename,input_names=inspect.getargspec(sss_filename)[0], output_names=['filename']))
-	bids_stim_filename = pe.Node(name='bids_stim_filename', interface=util.Function(function=bids_naming,input_names=inspect.getargspec(bids_naming)[0], output_names=['filename']))
```
Comment (TheChymera) [inline samri/pipelines/preprocess.py:182]: Why are you deleting this? We need to output a "bids_stim_file" which has been adjusted to reflect the dummy scan enforcement (and this stimulation summary file needs a name)

**Review by TheChymera — COMMENTED** [review]
This PR deletes a lot of important functionality from the workflow. I suspect this is because you had not pulled from master for a very long time.

When I said I had not changed anything except the `report`and `plotting` modules I was referring to the last week only. There have been a few changes to the pipeline since you implemented BIDS support.

Particularly the functionality of 06737a4ffcad92ec1e6ea00691c29b9cd8410cd2 is impacted negatively by this PR, though I expect there may be many other things not obvious just from reading the code. Further, this PR would not work since you are importing the `write_events_file` function, which no longer exists.

I would really recommend you pull the latest master, and than run a diff (you can use meld, which is a lot more intuitive than vimdiff) against this file and go through it line by line, introducing only the stuff that particularly addresses the issue of making the bandpass optional.

File: samri/pipelines/preprocess.py (hunk 1 above)
Comment (damaggu) [reply samri/pipelines/preprocess.py]: fixed

File: samri/pipelines/preprocess.py (hunk 2 above)
Comment (damaggu) [reply samri/pipelines/preprocess.py]: updated

File: samri/pipelines/preprocess.py (hunk 3 above)
Comment (damaggu) [reply samri/pipelines/preprocess.py]: updated

File: samri/pipelines/preprocess.py (hunk 4 above)
Comment (damaggu) [reply samri/pipelines/preprocess.py:182]: to cleanup and reduce workflow size, get_f_scan does now output that as well, already tested

**Pull request merged**